From 6a67dd4512a205a195821452f5db56e3658714e3 Mon Sep 17 00:00:00 2001
From: Kathiravan Thirumoorthy <quic_kathirav@quicinc.com>
Date: Tue, 13 Aug 2024 15:35:11 +0530
Subject: [PATCH] i2c: qcom-geni: update the i2c_clk_fld table for 32MHz

I2Cs present on the IPQ5424 supports 32MHz, not 19.2MHz. So update the
i2c_clk_fld table for the 32MHz.

Change-Id: I47f96ad88839a169b84df60ec8ef77f2616bab45
Signed-off-by: Kathiravan Thirumoorthy <quic_kathirav@quicinc.com>
---
 drivers/i2c/busses/i2c-qcom-geni.c | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/drivers/i2c/busses/i2c-qcom-geni.c b/drivers/i2c/busses/i2c-qcom-geni.c
index a9efd322df0e..85edb2fd8bcf 100644
--- a/drivers/i2c/busses/i2c-qcom-geni.c
+++ b/drivers/i2c/busses/i2c-qcom-geni.c
@@ -145,11 +145,21 @@ struct geni_i2c_clk_fld {
  * time of full period of SCL: t_cycle = (t_cycle_cnt * clk_div) / source_clock
  * clk_freq_out = t / t_cycle
  * source_clock = 19.2 MHz
+ *
+ * static const struct geni_i2c_clk_fld geni_i2c_clk_map[] = {
+ * 	{KHZ(100), 7, 10, 11, 26},
+ * 	{KHZ(400), 2,  5, 12, 24},
+ * 	{KHZ(1000), 1, 3,  9, 18},
+ * };
+*/
+
+/*
+ * source_clock = 32 MHz
  */
 static const struct geni_i2c_clk_fld geni_i2c_clk_map[] = {
-	{KHZ(100), 7, 10, 11, 26},
-	{KHZ(400), 2,  5, 12, 24},
-	{KHZ(1000), 1, 3,  9, 18},
+	{KHZ(100), 7, 14, 18, 40},
+	{KHZ(400), 4,  3, 11, 20},
+	{KHZ(1000), 4, 3,  6, 15},
 };
 
 static int geni_i2c_clk_map_idx(struct geni_i2c_dev *gi2c)
-- 
2.34.1

